### ISSUE TITLE
Enhance burn::tensor::module::attention() with scale, float mask, and softcap

### ISSUE THREAD

**Issue by antimora** [tracel-ai/burn#4475]
## Context

`burn::tensor::module::attention()` currently computes `softmax(QK^T / sqrt(d)) * V` with an optional bool mask. This is the right abstraction for backend-dispatched SDPA (enabling flash attention on CubeCL), but it lacks a few features needed to fully cover common attention patterns.

In [burn-onnx](https://github.com/tracel-ai/burn-onnx), we use `burn::tensor::module::attention()` for the ONNX Attention operator when possible, but fall back to manual tensor ops for three missing features. Adding them to the burn API would let more models benefit from flash attention automatically.

### 1. Custom scale parameter

**Current**: hardcoded `1/sqrt(d_k)` where `d_k` is the last dimension of Q.
**Requested**: `scale: Option<f64>` parameter, defaulting to `1/sqrt(d_k)` when `None`.

This is standard in other frameworks (PyTorch's `F.scaled_dot_product_attention` has a `scale` parameter). Some models use non-standard scaling.

We currently work around this by pre-scaling Q (`Q' = scale * sqrt(d) * Q`) so that `Q'K^T / sqrt(d) = scale * QK^T`, but native support would be cleaner and work correctly with flash attention kernels that may optimize the scaling differently.

### 2. Float additive mask (attention bias)

**Current**: only `Option<Tensor<B, 4, Bool>>` (positions set to `-inf`).
**Requested**: also accept a float tensor that gets **added** to attention scores before softmax.

Float masks are used for:
- ALiBi (Attention with Linear Biases)
- Relative position biases (T5, Swin Transformer)
- ONNX Attention's `attn_mask` when provided as float type

PyTorch's SDPA supports both bool and float masks via the same `attn_mask` parameter.

### 3. Softcap

**Current**: not supported.
**Requested**: `softcap: Option<f64>` parameter. When provided, applies `softcap * tanh(scores / softcap)` after QK matmul (and mask addition) but before softmax.

Used in Gemma-2 and related models. See [Gemma-2 blog post](https://huggingface.co/blog/gemma2#soft-capping-and-attention-implementations).

## Proposed signature

```rust
pub fn attention<B: Backend>(
    query: Tensor<B, 4>,
    key: Tensor<B, 4>,
    value: Tensor<B, 4>,
    mask: Option<Tensor<B, 4, Bool>>,
    // new parameters:
    scale: Option<f64>,
    attn_bias: Option<Tensor<B, 4>>,
    softcap: Option<f64>,
) -> Tensor<B, 4>
```

Or alternatively, an `AttentionOptions` struct to keep the API clean:

```rust
pub struct AttentionOptions<B: Backend> {
    pub scale: Option<f64>,
    pub attn_bias: Option<Tensor<B, 4>>,
    pub softcap: Option<f64>,
}
```

## Priority

From burn-onnx's perspective:
1. **Float additive mask** - most impactful, covers ALiBi and many real-world models
2. **Custom scale** - we have a workaround but native support is cleaner
3. **Softcap** - niche (Gemma-2 family) but increasingly common

## References

- ONNX Attention spec (opset 23+): https://onnx.ai/onnx/operators/onnx__Attention.html
- PyTorch SDPA: https://pytorch.org/docs/stable/generated/torch.nn.functional.scaled_dot_product_attention.html
- burn-onnx issue: https://github.com/tracel-ai/burn-onnx/issues/14
